<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="Test Checkout and Payment"
           tests="4"
           failures="2"
           errors="0"
           skipped="0"
           time="46.910">
    <properties></properties>

            <testcase
                name="tc0019_product_checkout_successfully"
                time="4.150"
                classname="com.ecomm.alexandnova.zenobiamayfield.testcases.PaymentTest">

            <failure type="org.openqa.selenium.StaleElementReferenceException" message="stale element reference: element is not attached to the page document
  (Session info: chrome=103.0.5060.53)
For documentation on this error, please visit: https://selenium.dev/exceptions/#stale_element_reference
Build info: version: &apos;4.0.0&apos;, revision: &apos;3a21814679&apos;
System info: host: &apos;DESKTOP-93CMEQ7&apos;, ip: &apos;10.0.0.132&apos;, os.name: &apos;Windows 10&apos;, os.arch: &apos;amd64&apos;, os.version: &apos;10.0&apos;, java.version: &apos;1.8.0_202&apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [8ac8f89e855a469440988d6b4b6acbc0, getElementText {id=5524f407-57cb-4b8b-84b8-6dfd956ee2bf}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 103.0.5060.53, chrome: {chromedriverVersion: 103.0.5060.53 (a1711811edd7..., userDataDir: C:\Users\Owner\AppData\Loca...}, goog:chromeOptions: {debuggerAddress: localhost:55719}, javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: WINDOWS, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:55719/devtoo..., se:cdpVersion: 103.0.5060.53, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Element: [[ChromeDriver: chrome on WINDOWS (8ac8f89e855a469440988d6b4b6acbc0)] -&gt; xpath: /html/body/div/div/div/main/div[1]/div/form/div[1]/div[2]/div[2]/div/p]
Session ID: 8ac8f89e855a469440988d6b4b6acbc0">
                <![CDATA[org.openqa.selenium.StaleElementReferenceException: stale element reference: element is not attached to the page document
  (Session info: chrome=103.0.5060.53)
For documentation on this error, please visit: https://selenium.dev/exceptions/#stale_element_reference
Build info: version: '4.0.0', revision: '3a21814679'
System info: host: 'DESKTOP-93CMEQ7', ip: '10.0.0.132', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_202'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [8ac8f89e855a469440988d6b4b6acbc0, getElementText {id=5524f407-57cb-4b8b-84b8-6dfd956ee2bf}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 103.0.5060.53, chrome: {chromedriverVersion: 103.0.5060.53 (a1711811edd7..., userDataDir: C:\Users\Owner\AppData\Loca...}, goog:chromeOptions: {debuggerAddress: localhost:55719}, javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: WINDOWS, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:55719/devtoo..., se:cdpVersion: 103.0.5060.53, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Element: [[ChromeDriver: chrome on WINDOWS (8ac8f89e855a469440988d6b4b6acbc0)] -> xpath: /html/body/div/div/div/main/div[1]/div/form/div[1]/div[2]/div[2]/div/p]
Session ID: 8ac8f89e855a469440988d6b4b6acbc0
                sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)java.lang.reflect.Constructor.newInstance(Constructor.java:423)org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:200)org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:133)org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:53)org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:184)org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:164)org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:139)org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:559)org.openqa.selenium.remote.RemoteWebElement.execute(RemoteWebElement.java:251)org.openqa.selenium.remote.RemoteWebElement.getText(RemoteWebElement.java:201)com.ecomm.alexandnova.zenobiamayfield.pages.PaymentInfoPage.getErrorMessage(PaymentInfoPage.java:91)com.ecomm.alexandnova.zenobiamayfield.testcases.PaymentTest.tc0019_product_checkout_successfully(PaymentTest.java:155)sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)java.lang.reflect.Method.invoke(Method.java:498)org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)org.testng.internal.Invoker.invokeMethod(Invoker.java:583)org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)org.testng.TestRunner.privateRun(TestRunner.java:648)org.testng.TestRunner.run(TestRunner.java:505)org.testng.SuiteRunner.runTest(SuiteRunner.java:455)org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)org.testng.SuiteRunner.run(SuiteRunner.java:364)org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)org.testng.TestNG.runSuitesLocally(TestNG.java:1137)org.testng.TestNG.runSuites(TestNG.java:1049)org.testng.TestNG.run(TestNG.java:1017)com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
]]>
            </failure>
    </testcase>
            <testcase
                name="tc0018_checkout_mandatory_fields_validation"
                time="4.597"
                classname="com.ecomm.alexandnova.zenobiamayfield.testcases.PaymentTest">

            <failure type="java.lang.AssertionError" message="expected [Your payment details couldn’t be verified. Check your card details and try again.] but found [Your payment details couldn&apos;t be verified. Check your card details and try again.]">
                <![CDATA[java.lang.AssertionError: expected [Your payment details couldn’t be verified. Check your card details and try again.] but found [Your payment details couldn't be verified. Check your card details and try again.]
                org.testng.Assert.fail(Assert.java:96)org.testng.Assert.failNotEquals(Assert.java:776)org.testng.Assert.assertEqualsImpl(Assert.java:137)org.testng.Assert.assertEquals(Assert.java:118)org.testng.Assert.assertEquals(Assert.java:453)org.testng.Assert.assertEquals(Assert.java:463)com.ecomm.alexandnova.zenobiamayfield.testcases.PaymentTest.tc0018_checkout_mandatory_fields_validation(PaymentTest.java:128)sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)java.lang.reflect.Method.invoke(Method.java:498)org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)org.testng.internal.Invoker.invokeMethod(Invoker.java:583)org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)org.testng.TestRunner.privateRun(TestRunner.java:648)org.testng.TestRunner.run(TestRunner.java:505)org.testng.SuiteRunner.runTest(SuiteRunner.java:455)org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)org.testng.SuiteRunner.run(SuiteRunner.java:364)org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)org.testng.TestNG.runSuitesLocally(TestNG.java:1137)org.testng.TestNG.runSuites(TestNG.java:1049)org.testng.TestNG.run(TestNG.java:1017)com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
]]>
            </failure>
    </testcase>


            <testcase
                name="tc0016_verify_discount_code_at_checkout_test"
                time="19.868"
                classname="com.ecomm.alexandnova.zenobiamayfield.testcases.PaymentTest">
            </testcase>
            <testcase
                name="tc0017_payment_method_check"
                time="18.295"
                classname="com.ecomm.alexandnova.zenobiamayfield.testcases.PaymentTest">
            </testcase>
</testsuite>
